restricted_site_access.php

```diff
@@ -58,6 +58,22 @@ public static function _add_actions() {
 		add_action( 'wpmu_new_blog', array( __CLASS__, 'set_defaults' ), 10, 6 );
 		add_action( 'admin_enqueue_scripts', array( __CLASS__, 'enqueue_admin_script' ) );
 		add_action( 'wp_ajax_rsa_notice_dismiss', array( __CLASS__, 'ajax_notice_dismiss' ) );
+
+		add_filter( 'restricted_site_access_is_restricted', array( __CLASS__, 'handle_constants' ) );
```

Comment:
we may want to consider hooking this late (eg priority 99) to overwrite any use of the filter in themes/plugins. My expectation is that the constant would override all other settings and if a theme or plugin added their own filter it might not.